File: frontend/lib/models/validation_result.dart
import 'cnab_record.dart';

class ValidationResult {
  final String fileName;
  final String status;
  final String? format;
  final String? type;
  final int totalLines;
  final int errorCount;
  final List<CnabRecord> records;
  final Map<String, String>? originalLines;

  ValidationResult({
    required this.fileName,
    required this.status,
    this.format,
    this.type,
    required this.totalLines,
    required this.errorCount,
    required this.records,
    this.originalLines,
  });

  factory ValidationResult.fromJson(Map<String, dynamic> json) {
    Map<String, String>? origLines;
    if (json['originalLines'] != null && json['originalLines'] is Map) {
      origLines = Map<String, String>.from(
        (json['originalLines'] as Map).map(
          (k, v) => MapEntry(k.toString(), v.toString()),
        ),
      );
    }
    
    return ValidationResult(
      fileName: json['fileName'] as String,
      status: json['status'] as String,
      format: json['format'] as String?,
      type: json['type'] as String?,
      totalLines: json['totalLines'] as int,
      errorCount: json['errorCount'] as int,
      records: (json['records'] as List)
          .map((r) => CnabRecord.fromJson(r as Map<String, dynamic>))
          .toList(),
      originalLines: origLines,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fileName': fileName,
      'status': status,
      'format': format,
      'type': type,
      'totalLines': totalLines,
      'errorCount': errorCount,
      'records': records.map((r) => r.toJson()).toList(),
      'originalLines': originalLines,
    };
  }

  bool get isValid => status == 'VALID';

  ValidationResult copyWith({
    String? fileName,
    String? status,
    String? format,
    String? type,
    int? totalLines,
    int? errorCount,
    List<CnabRecord>? records,
    Map<String, String>? originalLines,
  }) {
    return ValidationResult(
      fileName: fileName ?? this.fileName,
      status: status ?? this.status,
      format: format ?? this.format,
      type: type ?? this.type,
      totalLines: totalLines ?? this.totalLines,
      errorCount: errorCount ?? this.errorCount,
      records: records ?? this.records,
      originalLines: originalLines ?? this.originalLines,
    );
  }
}

File: backend/src/main/java/br/com/cnabvalidador/model/ValidationResult.java
package br.com.cnabvalidador.model;

import java.util.List;
import java.util.Map;

public class ValidationResult {
    private String fileName;
    private String status;
    private String format;
    private String type;
    private int totalLines;
    private int errorCount;
    private List<CnabRecord> records;
    private Map<String, String> originalLines;

    public ValidationResult() {
    }

    public ValidationResult(String fileName, String status, String format, String type,
                           int totalLines, int errorCount, List<CnabRecord> records) {
        this.fileName = fileName;
        this.status = status;
        this.format = format;
        this.type = type;
        this.totalLines = totalLines;
        this.errorCount = errorCount;
        this.records = records;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getFormat() {
        return format;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getTotalLines() {
        return totalLines;
    }

    public void setTotalLines(int totalLines) {
        this.totalLines = totalLines;
    }

    public int getErrorCount() {
        return errorCount;
    }

    public void setErrorCount(int errorCount) {
        this.errorCount = errorCount;
    }

    public List<CnabRecord> getRecords() {
        return records;
    }

    public void setRecords(List<CnabRecord> records) {
        this.records = records;
    }

    public Map<String, String> getOriginalLines() {
        return originalLines;
    }

    public void setOriginalLines(Map<String, String> originalLines) {
        this.originalLines = originalLines;
    }

    public boolean isValid() {
        return errorCount == 0;
    }
}
